### PR TITLE
20 by clicking on an edit button in the user account page the user is able to change his personal data

### PR DESCRIPTION
**Pull request by JoMuelle**
User should be able to change username and gender
No design yet
Added .trim to edit and register page to stop users using same name with spaces

### CODE REVIEW

File: src/components/routing/routers/AppRouter.js [hunk 1]
```diff
@@ -44,7 +45,11 @@ const AppRouter = () => {
             <UserPage/>
           </UserGuard>
         </Route>
-
+        <Route exact path={"/users/:id/edit"}>
```
Comment (daveking7) [inline src/components/routing/routers/AppRouter.js:48]: image
Something isn't right here

File: src/components/views/EditUserPage.js [hunk 2]
```diff
@@ -1,5 +1,128 @@
...
+    {value: 'MALE', label: 'Male'},
+    {value:  'FEMALE', label: 'Female'},
+    {value: 'OTHER', label: 'Other'}
+]
+
+
+const EditUserPage = () =>{
+    const history = useHistory();
+    const[user, setUser] = useState(null)
+    const [gender, setGender] = useState(null)
```
Comment (daveking7) [inline src/components/views/EditUserPage.js:43]: Do we really need gender and username states?

File: src/components/views/EditUserPage.js [hunk 3]
```diff
@@ -1,5 +1,128 @@
...
+                     </Button>
+                 </div>
+             </div>
+        )
+    }
+
+    const doEdit = async () => {
+        try {
+            localStorage.removeItem('errorMessage');
+            setUsername(username.trim())
```
Comment (daveking7) [inline src/components/views/EditUserPage.js:99]: null does not have a trim() method (error when press Save without new username)

File: src/components/views/UserPage.js [hunk 4]
```diff
@@ -104,6 +105,15 @@ const UserPage = () =>{
         </CardButton>
     }
 
+    const goToEdit = async () =>{
+        try {
+            history.push('/users/:id/edit')
```
Comment (daveking7) [inline src/components/views/UserPage.js:110]: BUGGY

File: src/components/views/Registration.js [hunk 5]
```diff
@@ -57,11 +57,9 @@ const Registration = () => {
       localStorage.removeItem('errorMessage');
       // post the new user to the server
 
-      const requestBody = JSON.stringify({username, name, password, birthday, gender});
+      const requestBody = JSON.stringify({username:username.trim(), name, password, birthday, gender});
```
Comment (daveking7) [inline src/components/views/Registration.js:60]: Try registering a user with username: '     ' (empty spaces) and see what happens (it shouldn't, right?)

**Review by daveking7 — CHANGES_REQUESTED** [review]

File: src/components/views/EditUserPage.js [hunk 6]
```diff
@@ -1,5 +1,128 @@
...
+            // put the new username and/or Gender to the server
+
+            const requestBody = JSON.stringify({username:username.trim(), gender});
+            await api.put('/users/'+localStorage.getItem("id"), requestBody);
+
+            // Registration successfully worked --> navigate to the route /game in the GameRouter
```
Comment (daveking7) [inline src/components/views/EditUserPage.js:105]: Comment seems a bit off

File: src/components/views/EditUserPage.js (hunk 3 above)
Comment (JoMuelle) [reply src/components/views/EditUserPage.js]: Username is never null, the username is the old username, so if you dont select a new one, the old one just gets passed

File: src/components/routing/routers/AppRouter.js (hunk 1 above)
Comment (JoMuelle) [reply src/components/routing/routers/AppRouter.js]: Fixed

File: src/components/views/UserPage.js (hunk 4 above)
Comment (JoMuelle) [reply src/components/views/UserPage.js]: Same mistake as above, fixed

File: src/components/views/Registration.js (hunk 5 above)
Comment (JoMuelle) [reply src/components/views/Registration.js:78]: Error 401 when trying, Username is already taken. Will try to fix

File: src/components/views/EditUserPage.js (hunk 6 above)
Comment (JoMuelle) [reply src/components/views/EditUserPage.js]: corrected

File: src/components/views/Registration.js [hunk 7]
```diff
@@ -103,12 +111,13 @@ const Registration = () => {
         <h2 className="registration title"> Registration </h2>
           <FormField
             label="Username"
-            value={username}
+            value={username.trim()} //can't add spaces at start or end of username, with this it is impossible to enter ""as username
```
Comment (daveking7) [inline src/components/views/Registration.js:114]: Very nice!

**Review by daveking7 — APPROVED** [review]
Mergeable!

**Pull request merged**